### PR TITLE
[flink-cdc] kafka_sync_database supports table name mapping when prefix and postfix could not fit the need.

### LINKED ISSUE #4659: [Feature] kafka_sync_database supports table name mapping when prefix and postfix could not fit the need.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

We are in the middle of transfer from existing warehouse to paimon. There are some irregular tables, whose names are not only prefixed or postfixed with the source tables. Even though we could adapt the sql to prefix and postfix, but it will cost too much time. So, we need the table name mapping. And the mapping rule has a higher priority than prefix and postfix. 

### Solution

Add a new configuration option, named as table_mapping. And user could supply more than one map rule, like the following:
```shell
kafka_sync_database
...
--table_mapping src1=tgt1
--table_mapping src2=tgt2
...
--table_mapping srcn=tgtn
...
```
If there is one table named as src1, src2, ..., srcn, the sinked table name should be tgt1, tgt2, ... tgtn, even though prefix or postfix are configured too.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by JackeyLee007**
### Purpose

Linked issue: close #4659

kafka_sync_database could map the source table to target table directly, when prefix and postfix is not suitable.  Some irregular table names could exists, which could not fitted by only prefix and postfix.

### Documentation

Add a new configuration for kafka_sync_database, which named as table_mapping.
For example,
```
kafka_sync_database 
...
--table_mapping src1=tgt1
--table_mapping src2=tgt2
...
--table_mapping srcn=tgtn
```
The data in src1, src2, srcn will sink into tgt1, tgt2, tgtn seperately.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java [hunk 1]
```diff
@@ -58,4 +71,18 @@ public String convert(Identifier originIdentifier) {
                                 + originIdentifier.getObjectName();
         return convert(rawName);
     }
+
+    private Map<String, String> lowerMapKey(Map<String, String> map) {
+        int size = map == null ? 0 : map.size();
+        Map<String, String> lowerKeyMap = new HashMap(size);
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java:77]: new HashMap<>(size);

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java [hunk 2]
```diff
@@ -58,4 +71,18 @@ public String convert(Identifier originIdentifier) {
                                 + originIdentifier.getObjectName();
         return convert(rawName);
     }
+
+    private Map<String, String> lowerMapKey(Map<String, String> map) {
+        int size = map == null ? 0 : map.size();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java:76]: if map == null, then return "Collections.emptyMap()".
I think this is better.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** Tests for {@link TableNameConverter}. */
+public class TableNameConverterTest {
+    private Map<String, String> tableMapping;
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:29]: delete this

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Test
+    public void testConvertTableName() {
+        Map<String, String> tableMapping = new HashMap<>();
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:33]: new HashMap<>(1);

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        tableMapping.put("mapped_src", "mapped_TGT");
+        TableNameConverter caseConverter =
+                new TableNameConverter(true, true, "pre_", "_pos", tableMapping);
+        Assert.assertEquals(caseConverter.convert("mapped_SRC"), "mapped_TGT");
+
+        Assert.assertEquals(caseConverter.convert("unmapped_src"), "pre_unmapped_src_pos");
+
+        TableNameConverter incaseConverter =
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:41]: incaseConverter?
May be withoutCaseConverter?

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java (hunk 1 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java:77]: Sure.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java (hunk 3 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:29]: Sure

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java (hunk 4 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:33]: Sure

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java (hunk 2 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/TableNameConverter.java:76]: I'd like insist. :-)

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java (hunk 5 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/TableNameConverterTest.java:41]: incaseConverter --> noCaseConverter

**Review by JingsongLi — COMMENTED** [review]
Can you also update documentation?

**Comment by JackeyLee007** [comment]
Doc provided, but triggered some irrelevant UT failure.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**